pkg/logger/logger.go

```diff
@@ -57,6 +61,9 @@ func (l *Logger) Debug(msg string, fields ...zap.Field) {
 }
 
 func (l *Logger) InfoWithID(reqID string, msg string, fields ...zap.Field) {
+	if l.noopReqIDLogging {
+		return
+	}
```

Comment:
I think you possibly avoid these duplicated early returns by defining a new internal interface which has these `...WithID` functions and encapsulating that within the logger. One implantation would run the `Check` function and the other would no-op, but I'm not sure if it's worth it, honestly, so I'll leave it to you. The value might be too low.

Reply:
I would agree that we could interface the logger out and have a shim for no-ops.
I do personally think this is a bit over the top though.